.happy/terraform/modules/sfn/main.tf

```diff
@@ -22,20 +22,74 @@ resource "aws_sfn_state_machine" "state_machine" {
       },
       "ApplyDefaults": {
         "Type": "Pass",
-        "Next": "DownloadValidate",
+        "Next": "Download",
         "Parameters": {
           "args.$": "States.JsonMerge($.inputDefaults, $$.Execution.Input, false)"
         },
         "ResultPath": "$.withDefaults",
         "OutputPath": "$.withDefaults.args"
       },
-      "DownloadValidate": {
+      "Download": {
+        "Type": "Task",
+        "Resource": "arn:aws:states:::batch:submitJob.sync",
+        "Next": "Validate",
+        "Parameters": {
+          "JobDefinition":"${var.job_definition_arn}",
+          "JobName": "download",
+          "JobQueue.$": "$.job_queue",
+          "RetryStrategy": {
+            "Attempts": ${var.max_attempts},
+            "EvaluateOnExit": [
+              {
+                "Action": "EXIT",
+                "OnExitCode": "1"
+              },
+              {
+                "Action": "RETRY",
+                "OnExitCode": "*"
+              }
+            ]
+          },
+          "ContainerOverrides": {
+            "Environment": [
+              {
+                "Name": "DROPBOX_URL",
+                "Value.$": "$.url"
+              },
+              {
+                "Name": "DATASET_ID",
+                "Value.$": "$.dataset_id"
+              },
+              {
+                "Name": "COLLECTION_ID",
+                "Value.$": "$.collection_id"
+              },
+              {
+                "Name": "STEP_NAME",
+                "Value": "download"
```

Comment:
Asked this in the other PR, but just checking to make sure this is correct. `"Value"` vs `"Value.$"`.

Reply:
download is not a json object so `$`'s are not needed

Reply:
LOL great point